tests/testsuite/binary_name.rs

```diff
@@ -284,25 +312,13 @@ fn check_msg_format_json() {
         .file("src/main.rs", "fn main() { assert!(true) }")
         .build();
 
-    let output = r#"
-{
-    "reason": "compiler-artifact",
-    "package_id": "path+file:///[..]/foo#0.0.1",
-    "manifest_path": "[CWD]/Cargo.toml",
-    "target": "{...}",
-    "profile": "{...}",
-    "features": [],
-    "filenames": "{...}",
-    "executable": "[ROOT]/foo/target/debug/007bar[EXE]",
-    "fresh": false
-}
-
-{"reason":"build-finished", "success":true}
-"#;
-
     // Run cargo build.
     p.cargo("build --message-format=json")
         .masquerade_as_nightly_cargo(&["different-binary-name"])
-        .with_json(output)
+        .with_stdout_data(str![[r#"
+{"executable":"[ROOT]/foo/target/debug/007bar[EXE]","features":[],"filenames":"{...}","fresh":false,"manifest_path":"[ROOT]/foo/Cargo.toml","package_id":"path+[ROOTURL]/foo#0.0.1","profile":"{...}","reason":"compiler-artifact","target":"{...}"}
```

Comment:
This is something worth fixing in snapbox. We need a pretty-printed JSON lines support.

Reply:
Technically, this is no longer jsonlines.  It looks like cargo-test-support had a baked in policy of taking `expected` and using double-newlines to delimit pretty-printed json.

One trick I can look into is we could parse a json array and then convert that to jsonlines.

Reply:
Yeah I am aware of the double-newlines trick in cargo-test-support. I don't consider it as a block for this test file. For test suites like `tests/testsuite/message_format.rs` we should consider postponing the migrations.